packages/zambdas/src/update-provider/index.ts

```diff
@@ -0,0 +1,83 @@
+import { APIGatewayProxyResult } from 'aws-lambda';
+import { availability, createFhirClient } from '../shared';
+import { createZambdaFromSkeleton } from '../shared/zambdaSkeleton';
+import { ErrorCodes, ZambdaFunctionInput, ZambdaFunctionResponse, ZambdaInput } from '../types';
+import { Operation } from 'fast-json-patch';
+import { FormData } from '../types';
+
+export const index = async (input: ZambdaInput): Promise<APIGatewayProxyResult> => {
+  return createZambdaFromSkeleton(input, performUpdate);
+};
+
+interface UpdatePractitionerInput {
+  data: FormData;
+  practitionerId: string;
+}
+
+const performUpdate = async (input: ZambdaFunctionInput): Promise<ZambdaFunctionResponse> => {
+  const { body, secrets } = input;
+  const { practitionerId, data } = body as UpdatePractitionerInput;
+
+  // if (!practitionerId || typeof practitionerId !== 'string') {
+  //   console.error('Invalid practitionerId');
+  //   return { error: ErrorCodes.missingRequired };
+  // }
+
+  // if (!data || typeof data !== 'object') {
+  //   console.error('Invalid data');
+  //   return { error: ErrorCodes.missingRequired };
+  // }
+
+  const patchOperations: Operation[] = [
+    {
+      op: data.slug ? 'replace' : 'add',
```

Comment:
Also, in future we will probably delete the registration page, and when providers register in the beginning, they can probably add/update their data from the profile page.

Reply:
Ok so in the case where the practitioner resource currently has no slug, why would that matter for our handling of the input?

My provider could have no slug, but if I provide a slug to this function it will attempt to 'replace' my non-existant slug. Am I on the right track?

Reply:
Can you break out the possible scenarios and what you expect to happen (and if the function does that)?

Provider currently has undefined property
1. Input data provides undefined property
2. Input data provides defined property

Provider currently has defined property
1. Input data provides undefined property
2. Input data provides defined property

I'm thinking we're trying to base logic on the input that you're attributing to the current state of the provider in our database, and that doesn't feel correct.